### PR TITLE
Adding SHA Digest based bundle options to Makefiles & CLI

### PR DESCRIPTION
**Pull request by theishshah**
In flight PR for supporting SHA Digest based bundles based on the EP linked [here](https://github.com/operator-framework/enhancements/pull/107)

See also meta issue #5511 

cc: @ryantking

### CODE REVIEW

File: internal/cmd/operator-sdk/generate/bundle/bundle.go [hunk 1]
```diff
@@ -219,6 +219,13 @@ func (c bundleCmd) runManifests() (err error) {
 		}
 	}
 
+	// Resolved SHA Digests if necessary
+	if c.useImageDigests {
+		if err := c.genSHADigest(); err != nil {
+			return fmt.Errorf("error resolving SHA Digest: %v", err)
+		}
+	}
+
```
Comment (ryantking) [inline internal/cmd/operator-sdk/generate/bundle/bundle.go:228]: Can we remove this

File: internal/cmd/operator-sdk/generate/bundle/bundle.go [hunk 2]
```diff
@@ -287,3 +294,9 @@ func (c bundleCmd) runMetadata() error {
 
 	return bundleMetadata.GenerateMetadata()
 }
+
+// genSHADigest will generate a SHA Digest based bundle for any images needed
+func (c bundleCmd) genSHADigest() error {
+	c.println("Resolving SHA Digests")
+	return nil
+}
```
Comment (ryantking) [inline internal/cmd/operator-sdk/generate/bundle/bundle.go:302]: And this

File: internal/plugins/manifests/v2/init.go [hunk 3]
```diff
@@ -155,14 +155,26 @@ IMAGE_TAG_BASE ?= %[1]s/%[2]s
...
+# You can enable this value if you would like to use SHA Based Digests
+# To enable set flag to true
+USE_IMAGE_DIGESTS ?= false
+
+# BUNDLE_GEN_FLAGS is an additional mechanism for resolving running sdk generate bundle flags
+BUNDLE_GEN_FLAGS ?= ""
+
+ifeq ($(USE_IMAGE_DIGESTS) true)
+	BUNDLE_GEN_FLAGS += "--use-image-digests"
+endif
```
Comment (ryantking) [inline internal/plugins/manifests/v2/init.go:169]:
```suggestion

# BUNDLE_GEN_FLAGS are the flags passed to the operator-sdk generate bundle command
BUNDLE_GEN_FLAGS ?= "-q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)"

# USE_IMAGE_DIGESTS defines if images are resolved via tags or digests
# You can enable this value if you would like to use SHA Based Digests
USE_IMAGE_DIGESTS ?= false
ifeq ($(USE_IMAGE_DIGESTS), true)
	BUNDLE_GEN_FLAGS += " --use-image-digests"
endif
```

Has this been tested? What you originally had was missing a comma in the ifeq call. Regen with my changes then make sure `make bundle` works in the test project.

File: internal/plugins/manifests/v2/init.go [hunk 4]
```diff
@@ -155,14 +155,26 @@ IMAGE_TAG_BASE ?= %[1]s/%[2]s
...
 `
 
 	makefileBundleFragmentGo = `
 .PHONY: bundle
 bundle: manifests kustomize ## Generate bundle manifests and metadata, then validate generated files.
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
-	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS) -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
```
Comment (ryantking) [inline internal/plugins/manifests/v2/init.go:177]:
```suggestion
	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS)
```

**Review by ryantking — CHANGES_REQUESTED** [review]
Looks good, few minor adjustments

File: internal/plugins/manifests/v2/init.go (hunk 3 above)
Comment (theishshah) [reply internal/plugins/manifests/v2/init.go]:
```
memcached-operator/bin/kustomize build config/manifests | operator-sdk generate bundle -q --overwrite --version 0.0.1   --use-image-digests
```

^^ above tested output with sample memcached operator

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**